packages/protobuf/src/create-registry.ts

```diff
@@ -71,9 +48,55 @@ export function createRegistry(
     findService(typeName: string): ServiceType | undefined {
       return services[typeName];
     },
+    findExtensionFor(typeName: string, no: number): Extension | undefined {
+      return extensionsByExtendee.get(typeName)?.get(no) ?? undefined;
+    },
+    findExtension(typeName: string): Extension | undefined {
+      return extensionsByName.get(typeName) ?? undefined;
+    },
   };
+
+  function addType(type: MessageType | EnumType | ServiceType | Extension) {
+    if ("fields" in type) {
```

Comment:
I take it this cannot use `instanceof` because the types in the union are all interfaces (vs. classes or other nominal JS types)? Unfortunate...

Reply:
That's correct, and I agree 😢